stat/wilcoxontest/wilcoxon_hyp.go

```diff
@@ -0,0 +1,182 @@
+package wilcoxon
+
+import (
+	"gonum.org/v1/gonum/stat/distuv"
+	"math"
+	"sort"
+)
+
+func ensureDataConformance(x []float64, y []float64) {
+	if len(x) == 0 {
+		panic("x array length is zero")
+	}
+	if len(y) == 0 {
+		panic("y array length is zero")
+	}
+	if len(x) != len(y) {
+		panic("dimension mismatch in ensure data conformance")
+	}
+}
+
+func CalculateDifferences(x []float64, y []float64, z []float64) {
```

Comment:
I'd probably rename `z` into `dst` and move it as 1st argument.
(and document that exported function)

is it really needed to be exported?

Reply:
thanks @sbinet , for reviewing the code, I don't think this function needs to be exported and a bunch of functions also don't need to be exported, I am making them private in the upcoming PR, Will make the suggested changes wrt to naming and positions of the arguments. Thanks @sbinet